src/Aspire.Dashboard/Components/Controls/PropertyGrid.razor.cs

```diff
@@ -6,13 +6,81 @@
 
-public partial class PropertyGrid<TItem>
+/// <summary>
+/// Describes an name/value item to be displayed in a <see cref="PropertyGrid{TItem}"/>.
+/// </summary>
+/// <remarks>
+/// <para>
+/// Implement this interface to use as the <c>TItem</c> of a <see cref="PropertyGrid{TItem}"/> component.
+/// </para>
+/// <para>
+/// The property grid has two columns, bound to display strings <see cref="Name"/> and <see cref="Value"/>.
+/// </para>
+/// <para>
+/// The <see cref="IsValueSensitive"/> and <see cref="IsValueMasked"/> properties control masking behavior,
+/// which prevents sensitive data from being displayed in the UI without user interaction.
+/// </para>
+/// </remarks>
+public interface IPropertyGridItem
+{
+    /// <summary>
+    /// Gets the display name of the item.
+    /// </summary>
+    string? Name { get; }
+
+    /// <summary>
+    /// Gets the display value of the item.
+    /// </summary>
+    string? Value { get; }
+
+    /// <summary>
+    /// Gets whether this item's value is sensitive and should be masked.
+    /// </summary>
+    /// <remarks>
+    /// Default implementation returns <see langword="false"/>.
+    /// </remarks>
+    public bool IsValueSensitive => false;
+
+    /// <summary>
+    /// Gets and sets whether this item's value is masked in the UI by default.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Masking is a security and privacy feature that causes values to appear as asterisks or other
+    /// characters in the UI. This is useful for sensitive data like passwords or API keys.
+    /// The user may choose to reveal the value by toggling the mask.
+    /// </para>
+    /// <para>
+    /// Only used when <see cref="IsValueSensitive"/> is <see langword="true"/>. Otherwise this property
+    /// is ignored.
+    /// </para>
+    /// </remarks>
+    public bool IsValueMasked { get => false; set => throw new NotImplementedException(); }
+
+    /// <summary>
+    /// Gets whether this item matches a filter string.
+    /// </summary>
+    /// <remarks>
+    /// Default implementation returns <see langword="false"/>.
+    /// </remarks>
+    /// <param name="filter">The search text to match against.</param>
+    /// <returns><see langword="true"/> if this item matches the filter, otherwise <see langword="false"/>.</returns>
+    public bool MatchesFilter(string filter) => false;
+}
+
+public partial class PropertyGrid<TItem> where TItem : IPropertyGridItem
 {
     private static readonly RenderFragment<TItem> s_emptyChildContent = _ => builder => { };
 
+    private static readonly GridSort<TItem> s_defaultNameSort = GridSort<TItem>.ByAscending(vm => vm.Name);
+    private static readonly GridSort<TItem> s_defaultValueSort = GridSort<TItem>.ByAscending(vm => vm.Value);
+
     [Parameter, EditorRequired]
     public IQueryable<TItem>? Items { get; set; }
 
+    [Parameter]
+    public Func<TItem, object?> ItemKey { get; init; } = static item => item.Name;
```

Comment:
What happens if the name isn't unique? There are a number of situations where we're kind of abusing the property grid to show things other than name/values. I wonder what will happen if someone accidently or indentionally adds items where the name isn't unique.

If there an exception thrown by the control? Or does it render but there are later display errors when updating.

I'm just wondering if ItemKey hasn't been changed, should validate the items names?

cc @vnbaaij

Reply:
Yes, this requires items to have unique names, which is a change in the case of duplicate names. There isn't much of a way around this as far as I can tell, shy of adding some other kind of identifier to the interface.

Duplicate keys causes exception:

> InvalidOperationException: More than one sibling of component 'Microsoft.FluentUI.AspNetCore.Components.FluentDataGridRow`1[TItem]' has the same key value, 'Property'. Key values must be unique.